### DISCUSSION TITLE
How to list workflows from a private repository

### DISCUSSION THREAD

**Discussion by Vadorequest** [community/community#27170] [Actions]
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-repository-workflows" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-repository-workflows" target="_blank" rel="noopener nofollow ugc">Actions - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#list-repository-workflows" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/free-pro-team@latest/rest/reference/actions</a></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Fetching this endpoint is working fine using a public repository. But how can I do this when using a private repository? There is no documentation regarding authentication and such.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The documentation you linked describes the requirement:</p>
<blockquote>
<p>If the repository is private you must use an access token with the <code>repo</code> scope. GitHub Apps must have the <code>actions:read</code> permission to use this endpoint.</p>
</blockquote>
<p>For generating an access token, see <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">Creating a personal access token</a>. Once you have it, you need to add an <code>Authorization</code> header to the HTTP request, with the content <code>token YOUR-TOKEN</code>.</p>

**Comment by Vadorequest** [comment]
<p>Thank you, that wasn’t obvious at first glance due to the lack of example. I’ll give it a try.</p>

**Comment by Vadorequest** [comment]
<p>Worked that way:</p>
<pre><code class="lang-auto">    const response = await fetch(GITHUB_API_LIST_PROJECT_WORKFLOWS, {
      headers: {
        Authorization: `token ${process.env.GITHUB_TOKEN}`,
      },
    });
</code></pre>
